app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/annotation/SymbolActivity.java

```diff
@@ -161,35 +166,86 @@ public boolean onOptionsItemSelected(MenuItem item) {
+    } else if (item.getItemId() == R.id.menu_action_anchor) {
+      symbol.setIconAnchor(Property.ICON_ANCHOR_BOTTOM);
+    } else if (item.getItemId() == R.id.menu_action_opacity) {
+      symbol.setIconOpacity(0.5f);
+    } else if (item.getItemId() == R.id.menu_action_offset) {
+      symbol.setIconOffset(new PointF(10.0f, 20.0f));
+    } else if (item.getItemId() == R.id.menu_action_text_anchor) {
+      symbol.setTextAnchor(Property.TEXT_ANCHOR_TOP);
+    } else if (item.getItemId() == R.id.menu_action_text_color) {
+      symbol.setTextColor(Color.WHITE);
+    } else if (item.getItemId() == R.id.menu_action_text_size) {
+      symbol.setTextSize(22f);
+    } else if (item.getItemId() == R.id.menu_action_z_index) {
+      symbol.setZIndex(0);
+    } else if (item.getItemId() == R.id.menu_action_animate) {
+      resetSymbol();
+      easeSymbol(symbol, new LatLng(6.687337, 0.381457), 180);
+      return true;
     } else {
-      if (item.getItemId() == R.id.menu_action_icon) {
-        symbol.setIconImage(MAKI_ICON_CAFE);
-      } else if (item.getItemId() == R.id.menu_action_rotation) {
-        symbol.setIconRotate(45.0f);
-      } else if (item.getItemId() == R.id.menu_action_text) {
-        symbol.setTextField("Hello world!");
-      } else if (item.getItemId() == R.id.menu_action_anchor) {
-        symbol.setIconAnchor(Property.ICON_ANCHOR_BOTTOM);
-      } else if (item.getItemId() == R.id.menu_action_opacity) {
-        symbol.setIconOpacity(0.5f);
-      } else if (item.getItemId() == R.id.menu_action_offset) {
-        symbol.setIconOffset(new PointF(10.0f, 20.0f));
-      } else if (item.getItemId() == R.id.menu_action_text_anchor) {
-        symbol.setTextAnchor(Property.TEXT_ANCHOR_TOP);
-      } else if (item.getItemId() == R.id.menu_action_text_color) {
-        symbol.setTextColor(Color.WHITE);
-      } else if (item.getItemId() == R.id.menu_action_text_size) {
-        symbol.setTextSize(22f);
-      } else if (item.getItemId() == R.id.menu_action_z_index) {
-        symbol.setZIndex(0);
-      } else {
-        return super.onOptionsItemSelected(item);
-      }
-      symbolManager.update(symbol);
+      return super.onOptionsItemSelected(item);
     }
+
+    symbolManager.update(symbol);
     return true;
   }
 
+  private void resetSymbol() {
+    symbol.setIconRotate(0.0f);
+    symbol.setGeometry(Point.fromLngLat(6.687337, 0.381457));
+    symbolManager.update(symbol);
+  }
+
+
+  private void easeSymbol(com.mapbox.mapboxsdk.plugins.annotation.Symbol symbol, final LatLng location, final float rotation) {
+    final LatLng originalPosition = symbol.getLatLng();
+    final float originalRotation = symbol.getIconRotate();
+    final boolean changeLocation = originalPosition.distanceTo(location) > 0;
+    final boolean changeRotation = originalRotation != rotation;
+    if (!changeLocation && !changeRotation) {
+      return;
+    }
+
+    ValueAnimator moveSymbol = ValueAnimator.ofFloat(0, 1).setDuration(5000);
```

Comment:
Reference to this animator should be kept and animation canceled when the new one starts (with a menu option click) or when the activity is stopped/finished. Otherwise, we are leaking and crashing. We can also harden this by refusing source updates after `AnnotationManager#onDestroy` call and logging an error (similarly to `NativeMapView` impl, this would prevent the crash but not the leak).

Reply:
I went with cancelling the animator before we stop rendering the map.